### ISSUE TITLE
Summary / dwca export of abundances in projects and collections

### ISSUE THREAD

**Issue by jiho** [ecotaxa/ecotaxa_dev#615]
The pseudo code to compute the abundance per sample is

```
GROUP BY sample, category
COUNT
```

where `category` is the list of category with at least one object classified in the current project/collection (after remapping #577).

The empty combinations sample × category should yield
- 0 in the summary export
- a DwC "absent" line and a concentration of 0 in dwca export:

> OBIS includes in their OBIS-ENV data format a DwC field called occurrenceStatus that can be either "present" or "absent". If the occurrenceStatus = "absent" no errors should emerge for a concentration (or any other biological measurement) = 0.

This should also be doable per subsample (i.e. acquisition).

**Comment by jiho** [comment]
In this case, the summary export could either be formatted in **long** format

```
sampleid	taxonid		count
sta1		Copepoda	10
sta1		Cnidaria	1
sta2		Copepoda	14
sta2		Cnidaria	0
```

or in **wide** format

```
sampleid	Copepoda	Cnidaria
sta1		10		1
sta2		14		0
```

whichever is easier.

**Comment by grololo06** [comment]
The **long** format is obviously easier, as the output _structure_ is fixed.

**Comment by grololo06** [comment]
⚠️ #601

**Comment by grololo06** [comment]
It's not clear what should be output when
> This should also be doable per subsample (i.e. acquisition).

especially when there are 0s around.
It's even more unclear when present issue is referenced in #616 and #617, especially #616 which can get data from 'subsample'.

**Comment by grololo06** [comment]
### Present state of code
The present summary export produces:
For this option:
![image](https://user-images.githubusercontent.com/20199778/173173290-40376223-f84d-4ad5-800d-aa56c1c1cab3.png)
A two-columns table with e.g.:

display_name | nbr
-- | --
Amphorellopsis | 3
Annelida | 1
Appendicularia | 1

For this option:
![image](https://user-images.githubusercontent.com/20199778/173173399-731abb3e-b84a-41e1-a496-9dbe41cdc28e.png)
orig_id | latitude | longitude | date | display_name | nbr
-- | -- | -- | -- | -- | --
emah_grp1_cote_10ml_4x | 0 | 0 | 2000-01-01 | Amphorellopsis | 2
emah_grp1_cote_10ml_4x | 0 | 0 | 2000-01-01 | Appendicularia | 1
emah_grp1_cote_10ml_4x | 0 | 0 | 2000-01-01 | Asterionellopsis glacialis<Asterionellopsis | 12

For this option:
![image](https://user-images.githubusercontent.com/20199778/173173450-062fa62c-a791-421b-8808-e695eace3bcc.png)

orig_id | display_name | nbr
-- | -- | --
flowcam_emah_grp1_cote_10ml_4x | Amphorellopsis | 2
flowcam_emah_grp1_cote_10ml_4x | Appendicularia | 1
flowcam_emah_grp1_cote_10ml_4x | Asterionellopsis glacialis<Asterionellopsis | 12
flowcam_emah_grp1_cote_10ml_4x | Chaetoceros<Chaetocerotales | 4
flowcam_emah_grp1_cote_10ml_4x | Ditylum brightwellii<Ditylum | 24

Remark: Due to https://github.com/ecotaxa/ecotaxa/issues/738, the third format is now wrong as there can be several acquisitions with same orig_id.

**Comment by jiho** [comment]
The export is per sampling unit (sample or subsample) and per taxon and should contain zeroes.

When there are no filters (export of whole project), the algorithm is:

1. compute the combinations of ALL sampling_units (from the samples or subsamples/acquisition table) and ALL categories (categories with at least one object classified in the project) 
2. compute counts of category per sampling unit with: `GROUP_BY sampling_unit, category; COUNT`
3. join 1 and 2 and add 0 for cases in 1 but not in 2.

When there are filters:
- if the filter is on categories, just replace "ALL categories" in step 1 by "the filtered categories". NB: the taxonomic filter currently allows only one category, but anticipate that there could be several : #156 
- if the filter is on samples and export is aggregated per sample, replace "ALL sampling_units" by "the filtered samples"
- if the filter is on samples and export is aggregated per _subsample_, replace "ALL sampling_units" by "the subsamples within the filtered samples"
- filtering subsamples is currently not possible but it would imply an export at subsample level (since aggregating at sample level from a subset of subsamples would be meaningless) and it would behave as filter on sample and export samples (second case above).

NB: the categories here must be understood as the categories still visible after remapping ( #601 ).

Examples:

Export of a project with samples 1,2,3 and categories A and B ; B is absent in sample 2; export at sample level.

| sample | category | n |
| ------- | --------- | - |
| 1 | A | 34 |
| 1 | B | 201 |
| 2| A | 21 |
| 2 | B | 0 |
| 3 | A | 15 |
| 3 | B | 302 |

Filtered for sample == 2

| sample | category | n |
| ------- | --------- | - |
| 2| A | 21 |
| 2 | B | 0 |

Filtered for sample == 2, category == B

| sample | category | n |
| ------- | --------- | - |
| 2 | B | 0 |

Filtered for sample == 2, export at subsample level (2 subsamples per sample, named d1, d2)

| sample | subsample | category | n |
| ------- | ----- | ---- | - |
| 2| d1 | A | 3 |
| 2| d2 | A | 18 |
| 2 | d1 | B | 0 |
| 2 | d2 | B | 0 |

**Comment by grololo06** [comment]
I just saw that it's in fact already possible to select several taxa, as the filter contains an option to pick a subtree from a given category (`taxochild`).

**Comment by jiho** [comment]
Indeed. So the taxonomic filter exists albeit not with the full flexibility expected. But the behaviour can be teste. Just expect that the selection of taxa may not be just a subtree of the taxonomic tree, it may be any random selection of taxa.

**Comment by grololo06** [comment]
API side implemented in v2.6.8.
